src/generic_core/remote-instance.ts

```diff
@@ -195,19 +189,20 @@ export var remoteProxyInstance :RemoteInstance = null;
           // TODO: Move the logic for resetting the onceSharerCreated promise inside
           // remote-connection.ts.
           this.connection_.resetSharerCreated();
-          this.onceSharerReadyForOffer_ = this.connection_.onceSharerCreated;
+          this.hasSeenOffer = true;
```

Comment:
I don't think we should rely on the client not being able to send an offer unless we are in NONE or TRYING_TO_SHARE_ACCESS.  It's possible that they send us an offer, then something goes wrong on their side, then they send us an offer again..  But in general I think its best to remove this code if possible, and rely on error handling in rtcToNet.  The main thing we want to do here is just create a new rtcToNet object each time we get an offer (which this.startShare should encapsulate)